osu.Game/Overlays/Notifications/ProgressNotification.cs

```diff
@@ -35,6 +39,27 @@ public float Progress
             }
         }
 
+        private bool cancellable = true;
+
+        /// <summary>
+        /// Whether this progress notification can be cancelled by user request.
+        /// </summary>
+        public bool Cancellable
+        {
+            get => cancellable;
+            set
+            {
+                if (cancellable == value)
+                    return;
+
+                if (state == ProgressNotificationState.Cancelled && value == false)
+                    throw new InvalidOperationException("Attempting to disable user cancelling while the notification is already cancelled.");
+
+                cancellable = value;
+                CloseButton.Enabled.Value = cancellable;
```

Comment:
can use compound assignment

```suggestion
                cancellable = CloseButton.Enabled.Value = cancellable;
```